application/backend/app/core/lifecycle.py

```diff
@@ -31,8 +33,14 @@ async def lifespan(app: FastAPI) -> AsyncGenerator[None]:
         logger.error("Failed to initialize database. Application cannot start.")
         raise RuntimeError("Database initialization failed")
 
+    active_pipeline_service = ActivePipelineService()
+    app.state.active_pipeline_service = active_pipeline_service
+
+    data_collector = DataCollector(active_pipeline_service=active_pipeline_service)
+    app.state.data_collector = data_collector
+
     # Initialize Scheduler
-    app_scheduler = Scheduler()
+    app_scheduler = Scheduler(active_pipeline_service=active_pipeline_service, data_collector=data_collector)
```

Comment:
Minor: it's arguably more practical to pass the whole `app.state` to `Scheduler`, so we don't have to update its constructor when there's a new class to provide as argument